### PR TITLE
add function to clean and install pip deps

### LINKED ISSUE #578: [CICD] Independent PIP env for multiple tests
follow-up of https://github.com/NVIDIA/NVFlare/pull/527/files

It would be nice to have a ENV context for each tests with in the same script.

So we can avoid potential conflicts.

### PR DESCRIPTION
**Pull request by YanxuanLiu**
fix #578 

1. maintain pip requirements with pipenv (already installed in docker image)
2. add args to shell scripts to enable running partial tests

### CODE REVIEW

**Review by pxLi — APPROVED** [review]
LGTM as a good start point. Please fill the PR desc well

File: ci/run_premerge_gpu.sh [hunk 1]
```diff
@@ -15,15 +15,57 @@
...
+integration_test() {
+    echo "Run integration test..."
+    init_pipenv requirements-dev.txt
+    testFolder="tests/integration_test"
+    rm -rf /tmp/snapshot-storage
+    pushd ${testFolder}
+    pipenv run ./run_integration_tests.sh
+    popd
+    rm -rf /tmp/snapshot-storage
+}
```
Comment (pxLi) [inline ci/run_premerge_gpu.sh:53]: please also add remove_pipenv as a good context patten to follow even there is only one test in this file for now

**Review by pxLi — CHANGES_REQUESTED** [review]
some small nit

File: ci/run_premerge_gpu.sh (hunk 1 above)
Comment (YanxuanLiu) [reply ci/run_premerge_gpu.sh:54]: added remove_pipenv in the end

**Pull request merged**